src/main/java/com/palantir/ri/ResourceIdentifier.java

```diff
@@ -140,7 +137,7 @@ public boolean equals(Object obj) {
             return false;
         }
         ResourceIdentifier other = (ResourceIdentifier) obj;
-        return resourceIdentifier.equals(other.resourceIdentifier);
+        return this.hashCode() == other.hashCode() && resourceIdentifier.equals(other.resourceIdentifier);
```

Comment:
Might be good to add a comment here explaining why we're doing this. Otherwise a future dev may remove this because it looks strange.

Reply:
good call, see https://github.com/palantir/resource-identifier/pull/334